src/diffusers/loaders.py

```diff
@@ -2178,6 +2201,65 @@ def unfuse_text_encoder_lora(text_encoder):
 
         self.num_fused_loras -= 1
 
+    def set_adapter(
+        self,
+        adapter_names: Union[List[str], str],
+        unet_weights: List[float] = None,
+        te_weights: List[float] = None,
+        te2_weights: List[float] = None,
```

Comment:
```suggestion
        te_weights: List[float] = None,
        text_encoder: Optional[PretrainedModel] = None,
```

Instead of having `te_weights` and `te2_weights` here, can we allow the user to optionally pass a text encoder and if the text encoder is None only then we default to `self.text_encoder`? A bit like how it's done here: https://github.com/huggingface/diffusers/pull/4903/files

Reply:
Or be a bit more explicit `set_adapter_for_text_encoder()`.